File: src/main/java/com/fluxpay/engine/presentation/dto/ApiResponse.java
package com.fluxpay.engine.presentation.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiResponse<T> {

    private final boolean isSuccess;
    private final String code;
    private final String message;
    private final T result;

    public static <T> ApiResponse<T> success(T result) {
        return new ApiResponse<>(true, "SUCCESS", "요청이 성공적으로 처리되었습니다.", result);
    }

    public static <T> ApiResponse<T> success(T result, String message) {
        return new ApiResponse<>(true, "SUCCESS", message, result);
    }

    public static <T> ApiResponse<T> error(String code, String message) {
        return new ApiResponse<>(false, code, message, null);
    }
}

File: src/main/kotlin/com/fluxpay/engine/presentation/dto/ApiResponse.kt
package com.fluxpay.engine.presentation.dto

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class ApiResponse<T>(
    val isSuccess: Boolean,
    val code: String,
    val message: String,
    val result: T? = null
) {
    companion object {
        fun <T> success(result: T? = null, message: String = "요청이 성공적으로 처리되었습니다."): ApiResponse<T> {
            return ApiResponse(
                isSuccess = true,
                code = "SUCCESS",
                message = message,
                result = result
            )
        }

        fun <T> error(code: String, message: String): ApiResponse<T> {
            return ApiResponse(
                isSuccess = false,
                code = code,
                message = message,
                result = null
            )
        }
    }
}
